### ISSUE TITLE
Raise unique exception (or `ValueError` subclass) on attempted initialization from non-unique values

### ISSUE THREAD

**Issue by flexatone** [brandtbucher/automap#7]
Greetings!

As you are aware, non-unique values given to an `AutoMap` raise a `ValueError`:

```
>>> AutoMap((3,1,3))
Traceback (most recent call last):
ValueError: 3
```

While this is generally good, other things can raise a value error on initialization, such as an iterator given to `AutoMap` that has un-evaluated code that might raise when evaluated.

```
>>> AutoMap(np.datetime64(d) for d in ("2021-01-88",))
Traceback (most recent call last):
ValueError: Day out of range in datetime string "2021-01-88"
```

The challenge is that `StaticFrame`, when calling `AutoMap`, catches `ValueErrors`, expecting them to be due to non-unique values, and delivers the user a detailed message explaining what happened. However, as shown above, `ValueErrors` can be raised for other reasons.

Might `AutoMap` raise a specialized exception for non-unique elements, such that clients can handle that more narrow exception appropriately? A subclass of `ValueError` that is exposed in the public API might be a nice, backwards compatible approach.

**Comment by brandtbucher** [comment]
Yep, I think this makes sense! PRs welcome, otherwise I should be able to get to it sometime this weekend.

(`automap` is overdue for a release anyways. There are a bunch of random incremental performance and memory improvements on `master` that I never got around to releasing!)

**Comment by flexatone** [comment]
Great to hear from you @brandtbucher ! Amusingly, I am not certain how to derive a subclass of `PyExc_ValueError`... I expect it might take a few lines in C. It will be great to get those performance improvements in too.

**Comment by flexatone** [comment]
Hi @brandtbucher ; I have a PR of this ready but CI is waiting your approval; also, can you review? https://github.com/brandtbucher/automap/pull/10

**Comment by brandtbucher** [comment]
Closed by #10.
